akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -1826,6 +1826,55 @@ final class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Grap
   def expand[U](extrapolate: function.Function[Out, java.util.Iterator[U]]): javadsl.Source[U, Mat] =
     new Source(delegate.expand(in ⇒ extrapolate(in).asScala))
 
+  /**
+   * Allows a faster downstream to progress of a slower upstream.
+   *
+   * This is achieved by introducing "extrapolated" elements - based on those from upstream - whenever downstream
+   * signals demand.
+   *
+   * Pressurize does not support [[akka.stream.Supervision#restart]] and [[akka.stream.Supervision#resume]].
+   * Exceptions from the `extrapolate` function will complete the stream with failure.
+   *
+   * '''Emits when''' downstream stops backpressuring, AND EITHER upstream emits OR initial element is present OR
+   * `extrapolate` is non-empty and applicable
+   *
+   * '''Backpressures when''' downstream backpressures or current `extrapolate` runs empty
+   *
+   * '''Completes when''' upstream completes and current `extrapolate` runs empty
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @param extrapolate takes the current upstream element and provides a sequence of "extrapolated" elements based
+   *                    on the original, to be emitted in case downstream signals demand
+   */
+  def pressurize[U >: Out](extrapolate: function.Function[U, java.util.Iterator[U]]): Source[U, Mat] =
+    new Source(delegate.pressurize[U](in ⇒ extrapolate(in).asScala.toStream))
+
+  /**
+   * Allows a faster downstream to progress of a slower upstream.
+   *
+   * This is achieved by introducing "extrapolated" elements - based on those from upstream - whenever downstream
+   * signals demand.
+   *
+   * Pressurize does not support [[akka.stream.Supervision#restart]] and [[akka.stream.Supervision#resume]].
+   * Exceptions from the `extrapolate` function will complete the stream with failure.
+   *
+   * '''Emits when''' downstream stops backpressuring, AND EITHER upstream emits OR initial element is present OR
+   * `extrapolate` is non-empty and applicable
+   *
+   * '''Backpressures when''' downstream backpressures or current `extrapolate` runs empty
+   *
+   * '''Completes when''' upstream completes and current `extrapolate` runs empty
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @param extrapolate takes the current upstream element and provides a sequence of "extrapolated" elements based
+   *                    on the original, to be emitted in case downstream signals demand
+   * @param initial the initial element to be emitted, in case upstream is able to stall the entire stream
+   */
+  def pressurize[U >: Out](extrapolate: function.Function[U, java.util.Iterator[U]], initial: U): Source[U, Mat] =
+    new Source(delegate.pressurize[U](in ⇒ extrapolate(in).asScala.toStream, Some(initial)))
```

Comment:
I would really want to avoid the Stream type, its memoization is not really something we need here, can just emit a strict collection

Reply:
Acknowledged, changing to `Iterator`, as with `expand`.

Reply:
Thanks :+